From 7d114622b8eb6744441eb503021c2e5547142664 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 06 Jun 2022 14:52:31 +0300
Subject: [PATCH] [SW-69581] gaudi3: ts: use psoc ts instead of cpu ts

In previous platforms the Arm CPU timestamp counter was used
as the system global timestamp counter.
But in Gaudi3 Arm CPU is not used, and even not present in many
images (onix/pldm).
Instead of the Arm CPU timestamp counter, PSOC timestamp counter
shell be used as the system global timestamp counter from now on.

Change-Id: Id092ea05a62d854c9f6150d03541a17f26bf6ce6
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 99907dd..b161b74 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -15,6 +15,7 @@
 #include <d0_psoc_reset_conf_regs.h>
 #include <d0_psoc_efuse_regs.h>
 #include <d0_psoc_uart_0_regs.h>
+#include <d0_psoc_timestamp_regs.h>
 #include <d0_parc_arc0_cfg_regs.h>
 #include <d0_psoc_i2c_s_regs.h>
 #include <d0_parc_int_aggr_i2c_slv_comb_regs.h>
@@ -43,7 +44,7 @@
 
 u32 hl_pll_timestamp_base_get(void)
 {
-	return mmD0_CPU_TIMESTAMP_CNTCR;
+	return mmD0_PSOC_TIMESTAMP_CNTCR;
 }
 
 unsigned int get_cs_pad_sel_flash(void)
